### PR TITLE
Prepare deploy sample implementation (public repo)

### PR DESCRIPTION
**Pull request by gigsanna**
### Overview of the plan

1. public repo should have a docker that can be used to build an app in a container
2. there's a separate private repo that will use this docker in it's GitHub action to build an app and deploy it to GCP (without exposing any secrets and/or internal infrastructure)

This is the first step, adding a docker to public example repo

### CODE REVIEW

File: lib/utils.ts [hunk 1]
```diff
@@ -16,3 +16,13 @@ export const envVarsPresent = () => {
...
+
+const privateEnvVars = {
+  APP_ENV: process.env.APP_ENV,
+  GIGS_PROJECT: process.env.GIGS_PROJECT,
+  GIGS_API_KEY: process.env.GIGS_API_KEY,
+}
+
+export function injectPrivateEnvVars() {
+  Object.assign(process.env, privateEnvVars)
+}
```
Comment (Plsr) [inline lib/utils.ts:28]: I don't really understand what this does and why we need it 🙈

Comment (Plsr) [reply lib/utils.ts:28]: I think I answered my question myself by missing to add this in the Tracking PR in the API Docs 😅

Comment (gigsanna) [reply lib/utils.ts:28]: Totally missed that question 😅 It's to ensure that relevant env variables are accessible when the project is built and running somewhere (or at least that's my understanding)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Production image, copy all the files and run next
+FROM base AS runner
+WORKDIR /app
+
+ARG APP_ENV
+ARG GIGS_PROJECT
+ARG GIGS_API_KEY
+
+ENV APP_ENV=$APP_ENV
```
Comment (3nvi) [inline Dockerfile:32]: double-checking we don't need the ENV variables before running `npm run build`. I'm pretty sure we don't but I just want to confirm

File: lib/utils.ts [hunk 3]
```diff
@@ -16,3 +16,13 @@ export const envVarsPresent = () => {
 
   return !!gigsProject && !!gigsAPIKey
 }
+
+const privateEnvVars = {
+  APP_ENV: process.env.APP_ENV,
+  GIGS_PROJECT: process.env.GIGS_PROJECT,
```
Comment (3nvi) [inline lib/utils.ts:22]: aren't these public vars?

**Review by 3nvi — APPROVED** [review]
Code LGTM! Haven't attempted to deploy it though

File: lib/utils.ts (hunk 3 above)
Comment (gigsanna) [reply lib/utils.ts:22]: Well, technically yes, I just stole and modified a little the function and  variable from portals 😅 I can rename them to public for clarity 👍 😄

File: Dockerfile (hunk 2 above)
Comment (gigsanna) [reply Dockerfile:32]: No, they're not affecting the build, just the running app 🙂

File: lib/utils.ts (hunk 3 above)
Comment (3nvi) [reply lib/utils.ts:22]: I'm indifferent. Don't worry about it 👍

**Pull request merged**